dmoj/problem_schema.yml

```diff
@@ -0,0 +1,142 @@
+#normal names
+type: map
+mapping:
+  "archive":
+    type: str
+    pattern: .+\.zip
+  "generator":
+    type: str
+  "checker":
+    type: any
+  "signature_grader":
+    type: map
+    mapping:
+      "entry":
+        type: str
+      "header":
+        type: str
+  "custom_judge":
+    type: str
+  "io_redirects":
+    type: map
+    mapping:
+      "DATA.in":
+        type: map
+        mapping:
+          "fd":
+            type: str
```

Comment:
`fd` is not necessarily a `str`. The options for it are:

* `stdout`
* `stdin`
* `stderr`
* any integer